### PR TITLE
csv-tui: init at 1.1

### PR DESCRIPTION
**Pull request by ottoblep**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by ottoblep** [comment]
I'm confused why the formatting is not satisfied here. I ran first nixpkgs-fmt then nixfmt, both didn't solve it.

**Comment by ottoblep** [comment]
I'm confused why the formatting is still not satisfied here. I ran first nixpkgs-fmt then nixfmt with both being insufficient.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 334839` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csv-tui</li>
  </ul>
</details>

**Comment by smancill** [comment]
If I remember correctly you need to add yourself as maintainer in a different commit.

**Comment by ottoblep** [comment]
> If I remember correctly you need to add yourself as maintainer in a different commit.

Thanks, i missed that.

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 334839` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csv-tui</li>
  </ul>
</details>

**Review by smancill — APPROVED** [review]

File: pkgs/by-name/cs/csv-tui/package.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    owner = "nathangavin";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-IRXLwZ2FHcCDmDVJ0xnV/4q+X2AFXPX/+Ph4Xxo3DyM=";
+  };
+
+  cargoHash = "sha256-wgeVcX0zSXffAuvKw2eKXC846WlC8F9UGMoxP3IXoLE=";
+
+  meta = with lib; {
+    description = "A terminal based csv editor which is designed to be not a ram hog like standard csv editors, but more useful than other text editors";
```
Comment (donovanglover) [inline pkgs/by-name/cs/csv-tui/package.nix:21]:
```suggestion
    description = "Terminal based csv editor designed to be not a ram hog like standard csv editors, but more useful than other text editors";
```
Refer to the `meta.description` rules here: https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/by-name/cs/csv-tui/package.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    homepage = "https://github.com/nathangavin/csv-tui";
+    license = licenses.unfree;
```
Comment (donovanglover) [inline pkgs/by-name/cs/csv-tui/package.nix:23]: Ask upstream to license this software so we can redistribute it. See: https://github.com/NixOS/nixpkgs/tree/master/pkgs#quick-start-to-adding-a-package

**Review by donovanglover — CHANGES_REQUESTED** [review]
Welcome to nixpkgs! The program seems relevant enough to include in nixpkgs. Just a few concerns below.

**Comment by donovanglover** [comment]
Additionally, I had to look through [the source code](https://github.com/nathangavin/csv-tui/blob/4dfd8a6e206a193ee31ab0e864466cd79ee9cc01/src/controller/defaultController.rs#L93-L338) to figure out what the keybinds were, so I don't think it's ready for general use yet.

File: pkgs/by-name/cs/csv-tui/package.nix (hunk 2 above)
Comment (smancill) [reply pkgs/by-name/cs/csv-tui/package.nix:23]: Agh, sorry I missed too many small details when checking the derivation.

**Comment by ottoblep** [comment]
> Additionally, I had to look through [the source code](https://github.com/nathangavin/csv-tui/blob/4dfd8a6e206a193ee31ab0e864466cd79ee9cc01/src/controller/defaultController.rs#L93-L338) to figure out what the keybinds were, so I don't think it's ready for general use yet.

All keybinds are annotated in the program itself except the arrow keys. I think it's actually very easy to use.

**Review by donovanglover — APPROVED** [review]
LGTM, thanks for making the necessary changes and getting upstream to add a license.

I've used other CSV TUIs in the past and this one does seem simpler and easier to use, so let's go with it

**Comment by smancill** [comment]
Result of `nixpkgs-review pr 334839` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>csv-tui</li>
  </ul>
</details>

**Pull request merged**